File: JeffSebastian/playground/src/main/java/nl/uva/se/ast/expression/logical/Or.java
package nl.uva.se.ast.expression.logical;

import nl.uva.se.ast.expression.Binary;
import nl.uva.se.ast.expression.Expression;
import nl.uva.se.visitor.Visitor;

public class Or extends Binary {

	public Or(int lineNumber, int offset, Expression left, 
			Expression right) {
		super(lineNumber, offset, left, right);		
	}

	@Override
	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

}

File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Model/Expressions/Binary/Or.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.AST.Model.Statements;
using UvA.SoftCon.Questionnaire.AST.Types;

namespace UvA.SoftCon.Questionnaire.AST.Model.Expressions.Binary
{
    public class Or : BinaryExpression
    {
        public override NodeType Type
        {
            get
            {
                return NodeType.Or;
            }
        }

        public Or(Operation operation, IExpression left, IExpression right, TextPosition position)
            : base(operation, left, right, position) { }

        public override T Accept<T>(IASTVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override DataType? GetType(IDictionary<string, DataType> symbolTable)
        {
            return DataType.Boolean;
        }
    }
}
